test/system/tools/fds/tests.py

```diff
@@ -107,24 +101,424 @@ def test_fast_discovery_closure(fast_discovery_tool):
+        if output_to_check in output:
+            # Success
+            exit_code = 0
+        else:
+            # Failure
+            print('STDOUT:')
+            print(output)
+            print('STDERR:')
+            print(err)
+            exit_code = 3
     else:
         # Failure
-        print('test_fast_discovery_closure FAILED')
         print('STDOUT:')
         print(output)
         print('STDERR:')
         print(err)
         exit_code = 3
 
+    return exit_code
+
+
+def test_fast_discovery_closure(fast_discovery_tool):
+    """Test that discovery command closes correctly."""
+    command = [fast_discovery_tool, '-i', '0']
+
+    output, err, exit_code = send_command(command)
+
+    EXPECTED_CLOSURE = "### Server shut down ###"
+
+    check_output(output, err, EXPECTED_CLOSURE, False)
+
+    sys.exit(exit_code)
+
+
+def test_fast_discovery_parse_XML_file_default_profile(fast_discovery_tool):
+    """Test that discovery command read XML default profile correctly."""
+
+    XML_file_path = 'test_xml_discovery_server.xml'
+    default_profile = XML_parse_profile(XML_file_path, "")
+
+    prefix = default_profile.getElementsByTagName('prefix')
+    PREFIX = prefix[0].firstChild.data
+
+    EXPECTED_SERVER_ADDRESS = []
+    udpv4 = default_profile.getElementsByTagName('udpv4')
+    for elem in udpv4:
+        address2 = elem.getElementsByTagName('address')[0].firstChild.data
+        port2 = elem.getElementsByTagName('port')[0].firstChild.data
+        if port2[0] == '0':
+            port2 = port2[1:]
+        EXPECTED_SERVER_ADDRESS.append("UDPv4:[" + address2 + "]:" + port2)
+
+    command = [fast_discovery_tool, '-x', XML_file_path]
+
+    output, err, exit_code = send_command(command)
+
+    EXPECTED_SERVER_ID = "Server GUID prefix: " + PREFIX.lower()
+    print(EXPECTED_SERVER_ID)
+
+    exit_code = check_output(output, err, EXPECTED_SERVER_ID, False)
+    if exit_code != 0:
+        sys.exit(exit_code)
+
+    for add in EXPECTED_SERVER_ADDRESS:
+        exit_code = check_output(output, err, add, False)
+        if exit_code != 0:
+            sys.exit(exit_code)
+
+    sys.exit(exit_code)
+
+
+def test_fast_discovery_parse_XML_file_URI_profile(fast_discovery_tool):
+    """Test that discovery command read XML profile using URI."""
+
+    XML_file_path = 'test_xml_discovery_server.xml'
+    uri_profile = XML_parse_profile(XML_file_path, "UDP_server_two")
+
+    prefix = uri_profile.getElementsByTagName('prefix')
+    PREFIX = prefix[0].firstChild.data
+
+    EXPECTED_SERVER_ADDRESS = []
+    udpv4 = uri_profile.getElementsByTagName('udpv4')
+    for elem in udpv4:
+        address2 = elem.getElementsByTagName('address')[0].firstChild.data
+        port2 = elem.getElementsByTagName('port')[0].firstChild.data
+        if port2[0] == '0':
+            port2 = port2[1:]
+        EXPECTED_SERVER_ADDRESS.append("UDPv4:[" + address2 + "]:" + port2)
+
+    command = [fast_discovery_tool, '-x', XML_file_path + '@UDP_server_two']
```

Comment:
If we change the profile position, this should be changed. Same elsewhere.

Reply:
Done in 6dd1ad92f